ote_sdk/ote_sdk/utils/argument_checks.py

```diff
@@ -0,0 +1,206 @@
+    """Function raises ValueError exception if string class name is not equal to expected"""
+    parameter_class_name = type(parameter).__name__
+    if (parameter_class_name in expected_class_names) is False:
+        raise ValueError(
+            f"Unexpected type of '{parameter_name}' parameter, expected: {expected_class_names}, actual: "
+            f"{parameter_class_name}"
+        )
+
+
+def check_dictionary_keys_values_type(
+    parameter, parameter_name, expected_key_class, expected_value_class
+):
+    """Function raises ValueError exception if dictionary keys or values have unexpected type"""
+    for key, value in parameter.items():
+        parameter_type = type(key)
+        if not isinstance(key, expected_key_class):
+            raise ValueError(
+                f"Unexpected type of nested '{parameter_name}' dictionary key, expected: {expected_key_class}, "
+                f"actual: {parameter_type}"
+            )
+        parameter_type = type(value)
+        if not isinstance(value, expected_value_class):
+            raise ValueError(
+                f"Unexpected type of nested '{parameter_name}' dictionary value, expected: {expected_value_class}, "
+                f"actual: {parameter_type}"
+            )
+
+
+def check_several_dictionaries_keys_values_type(parameter_name_expected_type: list):
+    """
+    Function checks if parameters dictionaries exist and raises ValueError exception if their key or value have
+    unexpected type
+    :param parameter_name_expected_type: list with tuples that contain dictionary parameter, name for exception message
+    and expected type
+    """
+    for (
+        parameter,
+        name,
+        expected_key_class,
+        expected_value_class,
+    ) in parameter_name_expected_type:
+        if parameter:
+            check_dictionary_keys_values_type(
+                parameter=parameter,
+                parameter_name=name,
+                expected_key_class=expected_key_class,
+                expected_value_class=expected_value_class,
+            )
+
+
+def check_that_string_not_empty(string: str, parameter_name: str):
+    """Function raises ValueError exception if string parameter is empty"""
+    if string == "":
+        raise ValueError(f"Empty string is specified as {parameter_name} parameter")
+
+
+def check_file_extension(
+    file_path: str, file_path_name: str, expected_extensions: list
+):
+    """Function raises ValueError exception if file has unexpected extension"""
+    file_extension = file_path.split(".")[-1].lower()
+    if file_extension not in expected_extensions:
+        raise ValueError(
+            f"Unexpected extension of {file_path_name} file. expected: {expected_extensions} actual: {file_extension}"
+        )
+
+
+def check_that_null_character_absents_in_path(file_path: str, file_path_name: str):
+    """Function raises ValueError exception if null character: '\0' is specified in path to file"""
+    if "\\0" in file_path:
```

Comment:
**Attention!** I think that there should be `"\0" in file_path` instead of `"\\0" in file_path`,
since `"\0"` is a string that consists of one char with code zero, whereas `"\\0"` is just a string that consists of two chars: `"\\"` and `"0"`
```suggestion
    if "\0" in file_path:
```

Reply:
Applied suggestion, thanks